### ISSUE TITLE
Inconsistent "OK to retry" judgments in Programming.Errors.md

### ISSUE THREAD

**Issue by nyh** [awsdocs/amazon-dynamodb-developer-guide#208]
In Programming.Errors.md it is described what "OK to retry" means, and then a judgment of "Yes" or "No" is given for each of the error types, without an explanation why and in some cases, this judgment seems wrong or inconsistent.

One example is `ItemCollectionSizeLimitExceededException`, for which the "OK to retry" is given as "YES". However, retrying the same operation cannot possibly succeed, because the limit is still in place. So why "YES"? True, if some parallel operations remove items we may go below the limit, so maybe this explains the YES? I don't think so... Consider `ConditionalCheckFailedException` - this one is marked NO. But there to, parallel requests may cause the condition to succeed, so a client may want to retry. In short, it's not clear why `ItemCollectionSizeLimitExceededException` and  `ConditionalCheckFailedException` should have a different "OK to retry" judgment.

Another interesting case is `UnrecognizedClientException`. If the client has a wrong key, it is pointless to retry - the key will still be wrong. And yet, this error is marked "YES". Does this suggest that there are cases where AWS may be unable to verify a key (e.g., because of a network problem) and will temporarily return `UnrecognizedClientException` instead of `Internal Server Error` (which I would have expected)? Does this mean a client should always retry when getting a `UnrecognizedClientException` because this possiblity? This is not just a theoretical issue - people have actually been troubled by this question, see https://github.com/boto/boto3/issues/509.

To summarize, I think the YES/NO judgments in this document should be carefully reviewed, and wherever the decision is not obvious - a longer text explaining *why* this decision was chosen would have been nice.

**Comment by joshbean** [comment]
Closing this issue or pull request in advance of archiving this repo. For more information about the decision to archive this repo (and others in the 'awsdocs' org), see [the announcement on the AWS News Blog](https://aws.amazon.com/blogs/aws/retiring-the-aws-documentation-on-github/).
